modules/caddyhttp/rewrite/caddyfile.go

```diff
@@ -110,3 +114,53 @@ func parseCaddyfileURI(h httpcaddyfile.Helper) (caddyhttp.MiddlewareHandler, err
 	}
 	return rewr, nil
 }
+
+func parseCaddyfileHandlePath(h httpcaddyfile.Helper) ([]httpcaddyfile.ConfigValue, error) {
+	if !h.Next() {
+		return nil, h.ArgErr()
+	}
+	if !h.NextArg() {
+		return nil, h.ArgErr()
+	}
+
+	// read the prefix to strip
+	path := h.Val()
+	if !strings.HasPrefix(path, "/") {
+		path = "/" + path
+	}
+
+	// the ParseSegmentAsSubroute function expects the cursor
+	// to be at the token just before the block opening,
+	// so we need to rewind because we already read past it
+	h.Reset()
+	h.Next()
```

Comment:
This is a bit funky, but aside from changing how `ParseSegmentAsSubroute` works to avoid the extra `h.Next()`, I'm not sure what to do.